lib/h3core.js

```diff
@@ -843,6 +895,91 @@ function h3Distance(origin, destination) {
     return H3.h3Distance(oLower, oUpper, dLower, dUpper);
 }
 
+/**
+ * Produces IJ coordinates for an H3 index anchored by an origin.
+ *
+ * - The coordinate space used by this function may have deleted
+ * regions or warping due to pentagonal distortion.
+ * - Coordinates are only comparable if they come from the same
+ * origin index.
+ * - Failure may occur if the index is too far away from the origin
+ * or if the index is on the other side of a pentagon.
+ * - This function is experimental, and its output is not guaranteed
+ * to be compatible across different versions of H3.
+ * @static
+ * @param  {H3Index} origin      Origin H3 index
+ * @param  {H3Index} destination H3 index for which to find relative coordinates
+ * @return {Object}              Coordinates as an `{i, j}` pair
+ * @throws {Error}               If the IJ coordinates cannot be calculated
+ */
+function experimentalH3ToLocalIj(origin, destination) {
+    const ij = C._malloc(SZ_COORDIJ);
+    const [oLower, oUpper] = h3IndexToSplitLong(origin);
+    const [dLower, dUpper] = h3IndexToSplitLong(destination);
+    const retVal = H3.experimentalH3ToLocalIj(oLower, oUpper, dLower, dUpper, ij);
+    const coords = readCoordIJ(ij);
+    C._free(ij);
+    // Return the pair, or throw if an error code was returned.
+    // Switch statement and error codes cribbed from h3-java's implementation.
+    switch (retVal) {
+        case 0:
+            return coords;
+        case 1:
```

Comment:
Nit: These error codes are named in the C code, could these names also be used here?

Reply:
Talked to @isaacbrodsky about this, they aren't actually named in C yet. We ought to have a way to communicate the semantic content of these errors from C, but looks like this is the best we can do at the moment.

Reply:
That's unexpected. I thought I was the one playing fast and loose with error code numbers and @isaacbrodsky was the voice of reason to make that clearer. :)

Reply:
This was done somewhat fast and loose in the Java bindings, which is where @nrabinowitz got the original code from. It's maybe not a good practice, since nobody knows what these codes mean.

I would like to add a proper error enum type in H3 v4 that can be consumed by the bindings (or maybe something else that better communicates the errors?) This would be part of changing the API to more consistently return errors.

We could introduce named constants here but I think it would be better to update the C code to have the named constants first.